mixed-content/generic/common.js

```diff
@@ -0,0 +1,238 @@
+function normalizePort(targetPort) {
+  var defaultPorts = [80, 443];
+  var isDefaultPortForProtocol = (defaultPorts.indexOf(targetPort) >= 0);
+
+  return (targetPort == "" || isDefaultPortForProtocol) ?
+          "" : ":" + targetPort;
+}
+
+function setAttributes(el, attrs) {
+  attrs = attrs || {}
+  for (var attr in attrs)
+    el[attr] = attrs[attr];
+}
+
+function createElement(tagName, attrs, parent) {
+  var el = document.createElement(tagName);
+  setAttributes(el, attrs);
+  if (parent && parent.appendChild)
+    parent.appendChild(el);
+
+  return el;
+}
+
+function createHelperIframe(name) {
+  return createElement("iframe", {"name": name}, document.body);
+}
+
+function queryIframe(url) {
+  createElement("iframe", {"src": url}, document.body);
+}
+
+function queryImage(url) {
+  createElement("img", {"src": url}, document.body)
+}
+
+function queryXhr(url) {
+  var xhr = new XMLHttpRequest();
+  xhr.open('GET', url, true);
+  xhr.responseType = "json";
+  xhr.send();
```

Comment:
Understood. However, for developers it's important to get cross-browser agreement about what the web-visible side-effects of mixed content blocking are for various resource types. That is, browser should behave as though there was a network error, which has certain impacts on events, etc. The tests should reflect that, regardless of the reality between browsers. Then we fix the browsers. :)